### DISCUSSION TITLE
Tailwind causing CLS in Next.js

### DISCUSSION THREAD

**Discussion by humet** [tailwindlabs/tailwindcss#11706] [Help]
I have a very unusual problem with a very specific edge case, however it's impacted our website quite a lot. 

I've noticed on slower connections that Tailwind will cause CLS on the `body` element when a page loads in a Next.js application. But... it only happens if a user is scrolling while the page is loading. 

To reproduce:

1) Clone this repo and install locally: https://github.com/Dexerto/with-tailwindcss-app
2) Build the project.
3) Start the local server.
4) Open the app in Chrome. Throttle the network connection to Fast 3G.
5) Reload or load the page again.
6) Scroll down while the page is reloading or as soon as it loads. 
7) Check CLS (We use the bookmarklet from [this page](https://tamethebots.com/blog-n-bits/web-perf-iv-a-new-hope))

With scrolling causes CLS:

<img width="1440" alt="Screenshot 2023-07-27 at 17 37 19" src="https://github.com/tailwindlabs/tailwindcss/assets/3963951/baa263ba-a456-4c60-b9ad-56d9eb38362d">

Without scrolling or faster connection there is no CLS:

<img width="1440" alt="Screenshot 2023-07-27 at 17 37 34" src="https://github.com/tailwindlabs/tailwindcss/assets/3963951/f48f3be7-cdd4-43e0-acf2-df0ee65555e7">

This is the layout shift object:

```
{
    "name": "",
    "entryType": "layout-shift",
    "startTime": 1347.2999999999884,
    "duration": 0,
    "value": 1,
    "hadRecentInput": false,
    "lastInputTime": 0,
    "sources": [
        {
            "previousRect": {
                "x": 8,
                "y": 0,
                "width": 1424,
                "height": 764,
                "top": 0,
                "right": 1432,
                "bottom": 764,
                "left": 8
            },
            "currentRect": {
                "x": 0,
                "y": 0,
                "width": 1440,
                "height": 764,
                "top": 0,
                "right": 1440,
                "bottom": 764,
                "left": 0
            }
        }
    ]
}
```

Just to note, I can't visually see and shift. The page loads with no movement of any element. When I remove the base and utilities styles I don't get any CLS, so my assumption is that it's Tailwind related. 

I understand this is a small edge case, but our users seem to be inpatient and scroll down while the page is loading! 

Does anybody know why this could happen?

**Comment by syedsimanta03** [comment]
I don't think itz just tailwind, maybe config issue

**Comment by mybearworld** [comment]
What is CLS?

**Comment by Emmerhaus** [comment]
I'm also encountering the same issue. Did you manage to fix it? @humet

**Comment by Planxnx** [comment]
found same problem
